tests/test_extensions/test_sklearn_extension/test_sklearn_extension.py

```diff
@@ -1254,101 +1273,259 @@ def test_seed_model_raises(self):
+        for i in (0, 1, 2, 4, 5):
+            self.assertTrue(np.any(y_hat_proba[:, i] != np.zeros(y_test.shape)))
+
+        # check user defined measures
+        fold_evaluations = collections.defaultdict(lambda: collections.defaultdict(dict))
+        for measure in user_defined_measures:
+            fold_evaluations[measure][0][0] = user_defined_measures[measure]
+
         # trace. SGD does not produce any
-        self.assertIsInstance(arff_tracecontent, list)
-        self.assertEqual(len(arff_tracecontent), 0)
+        self.assertIsNone(trace)
+
+        self._check_fold_timing_evaluations(fold_evaluations, num_repeats, num_folds,
+                                            task_type=task.task_type_id, check_scores=False)
 
-        fold_evaluations = collections.defaultdict(
-            lambda: collections.defaultdict(dict))
+    def test_run_model_on_fold_classification_2(self):
+        task = openml.tasks.get_task(7)
+        num_folds = 1
+        num_repeats = 1
+
+        X, y = task.get_X_and_y()
+        train_indices, test_indices = task.get_train_test_split_indices(
+            repeat=0, fold=0, sample=0)
+        X_train = X[train_indices]
+        y_train = y[train_indices]
+        X_test = X[test_indices]
+        y_test = y[test_indices]
+
+        pipeline = sklearn.model_selection.GridSearchCV(
+            sklearn.tree.DecisionTreeClassifier(),
+            {
+                "max_depth": [1, 2],
+            },
+        )
+        # TODO add some mocking here to actually test the innards of this function, too!
+        res = self.extension._run_model_on_fold(
+            model=pipeline,
+            task=task,
+            fold_no=0,
+            rep_no=0,
+            X_train=X_train,
+            y_train=y_train,
+            X_test=X_test,
+            classes=task.class_labels,
+        )
+
+        y_hat, y_hat_proba, user_defined_measures, trace = res
+
+        # predictions
+        self.assertIsInstance(y_hat, np.ndarray)
+        self.assertEqual(y_hat.shape, y_test.shape)
+        self.assertIsInstance(y_hat_proba, np.ndarray)
+        self.assertEqual(y_hat_proba.shape, (y_test.shape[0], 2))
+        np.testing.assert_array_almost_equal(np.sum(y_hat_proba, axis=1), np.ones(y_test.shape))
+        for i in (0, 1):
+            self.assertTrue(np.any(y_hat_proba[:, i] != np.zeros(y_test.shape)))
+
+        # check user defined measures
+        fold_evaluations = collections.defaultdict(lambda: collections.defaultdict(dict))
         for measure in user_defined_measures:
             fold_evaluations[measure][0][0] = user_defined_measures[measure]
 
+        # check that it produced and returned a trace object of the correct length
+        self.assertIsInstance(trace, OpenMLRunTrace)
+        self.assertEqual(len(trace.trace_iterations), 2)
+
         self._check_fold_timing_evaluations(fold_evaluations, num_repeats, num_folds,
```

Comment:
same note on `num_repeats` and `num_folds`

Reply:
Done.